test/ts/unit/collateralizer/runners/collateralize.ts

```diff
@@ -0,0 +1,186 @@
+
+// scenarios
+import {CollateralizeScenario, TestAccounts, TestContracts} from "./";
+
+// utils
+import { REVERT_ERROR } from "../../../test_utils/constants";
+
+// logs
+import { CollateralLocked } from "../../../logs/collateralized_contract";
+import {MockTokenTransferProxyContract} from "../../../../../types/generated/mock_token_transfer_proxy";
+
+export class CollateralizeRunner {
+    private contracts: TestContracts;
+    private accounts: TestAccounts;
+
+    constructor() {
+        this.testScenario = this.testScenario.bind(this);
+    }
+
+    public initialize(contracts: TestContracts, accounts: TestAccounts) {
+        this.contracts = contracts;
+        this.accounts = accounts;
+    }
+
+    public testScenario(scenario: CollateralizeScenario) {
+        let ATTACKER: string;
+        let COLLATERALIZER: string;
+        let MOCK_DEBT_KERNEL_ADDRESS: string;
+        let MOCK_TERMS_CONTRACT_ADDRESS: string;
+
+        let collateralizer: CollateralizerContract;
+        let mockDebtRegistry: MockDebtRegistryContract;
+        let mockCollateralToken: MockERC20TokenContract;
+        let mockTokenRegistry: MockTokenRegistryContract;
+        let mockTokenTransferProxy: MockTokenTransferProxyContract;
+
+        let txHash: string;
+
+        describe(scenario.description, () => {
+            before(async () => {
+                ATTACKER = this.accounts.ATTACKER;
+                COLLATERALIZER = this.accounts.COLLATERALIZER;
+                MOCK_DEBT_KERNEL_ADDRESS = this.accounts.MOCK_DEBT_KERNEL_ADDRESS;
+                MOCK_TERMS_CONTRACT_ADDRESS = this.accounts.MOCK_TERMS_CONTRACT_ADDRESS;
+
+                collateralizer = this.contracts.collateralizer;
+                mockDebtRegistry = this.contracts.mockDebtRegistry;
+                mockCollateralToken = this.contracts.mockCollateralToken;
+                mockTokenRegistry = this.contracts.mockTokenRegistry;
+                mockTokenTransferProxy = this.contracts.mockTokenTransferProxy;
+
+                await mockDebtRegistry.reset.sendTransactionAsync();
+                await mockCollateralToken.reset.sendTransactionAsync();
+                await mockTokenRegistry.reset.sendTransactionAsync();
+
+                // Mock the collateral token's presence in the token registry at
+                // the specified index
+                await mockTokenRegistry.mockGetTokenAddressByIndex.sendTransactionAsync(
+                    scenario.collateralTokenIndexInRegistry,
+                    mockCollateralToken.address,
+                );
+
+                // Mock collateral token's balance for collateralizer
+                await mockCollateralToken.mockBalanceOfFor.sendTransactionAsync(
+                    COLLATERALIZER,
+                    scenario.collateralTokenBalance,
+                );
+
+                // Mock collateral token's allowance for collateralizer
+                await mockCollateralToken.mockAllowanceFor.sendTransactionAsync(
```

Comment:
Shouldn't this allowance be to the token transfer proxy rather than to the collateralizer?

Reply:
Ah just saw the following statement that mocks an allowance to the proxy.

We should this allowance granting statement then, given that it's not contributing anything to our test cases.